### ISSUE TITLE
Supported Wwise version?

### ISSUE THREAD

**Issue by mbbmbbmm** [Wasted-Audio/hvcc#123]
Hi, I am trying to compile Pd-patches for Wwise, so I can use them in Godot 4.
I am on Windows 10 btw.
The current [Godot-Integration for Wwise](https://github.com/alessandrofama/wwise-godot-integration/) seems to require using the 2023.1.0 Beta version. This version however does not seem to like the output from hvcc. It seems to have issues with the newest SDK / build tools / platform tool set (?) 
Which is the highest Wwise version supported by hvcc at the moment?
Thank you!
Jan

**Comment by dromer** [comment]
To be honest it's pretty impossible for me to test Wwise as it has very poor Linux support.
I've been able to compile a Wwise plugin once, and that was with a fairly recent SDK version. However I had no way to confirm that it actually worked inside of an integration (like with Unity or something).

In the code the Wwise version is hardcoded on a pretty old version: https://github.com/Wasted-Audio/hvcc/blob/develop/hvcc/generators/c2wwise/c2wwise.py#L60
But you should be able to change that to whatever version you need.

Can you be a bit more specific with "does not seem to like the output"? Are there any specific errors?

**Comment by mbbmbbmm** [comment]
Yes, of course. When I try to open the solution in Visual Studio I get errors like:
The imported project "C:\Program Files (x86)\Audiokinetic\Wwise 2023.1.0.8256\SDK\source\Build\PropertySheets\Win32\Debug_vc140.props" was not found.
There only are higher number versions present in that directory, 160 and 170

(I'm pretty sure I have already updated the template to the correct Wwise version)

**Comment by dromer** [comment]
What do you mean with "open the solution"?

**Comment by mbbmbbmm** [comment]
So, when hvcc in command to compile the patch it generates several folders linux, resources, source, vs2015, xcode plus a build.json. The way I've understood it is to open the Hv_None_WwiseSourcePlugin.sln solution file located in the vs2015 folder with Visual Studio and build the proper plugin from there.

**Comment by dromer** [comment]
Oh sorry, I'm not familiar with Visual Studio nomenclature. So this "solution" file description has no meaning to me, haha.

Maybe try building the plugin without using Visual Studio? To be honest I cannot directly support this build system, but I'm happy to take contributions to improve it.

**Comment by mbbmbbmm** [comment]
Oh yeah, sorry. It is like a glorified makefile I guess. So I could switch to Linux and run the provided makefile instead and see what happens. Will do that later/tomorrow and report back. If that works I could try to build without Visual Studio on Windows instead. Though I don't have any experience with C or C++ unfortunately.

**Comment by dromer** [comment]
I heard people are still using this stuff for commercial products, so hopefully some of them will step in to help improve the state.
Sorry that I can't help any further, but Wwise isn't make it very easy to do so.

(to be honest I hope we can find a proper OSS solution for Godot integration)

**Comment by mbbmbbmm** [comment]
That would be awesome. The only thing I am aware of that doesn't use Wwise for Godot is 
[https://github.com/magogware/godot-audiostreampd](https://github.com/magogware/godot-audiostreampd)
But it is too slow, input is not recognized in time.

What would I have to change in the Linux Makefile to run it on Windows? I have downloaded the little [w64 devkit](https://github.com/skeeto/w64devkit) here but running the Linux file kind of expectedly gives errors..

**Comment by mbbmbbmm** [comment]
OK, cool, I've managed to figure it out (it seems, still have to test in Godot).
First of all I  upgraded to VS Community 2022 including the normal C++ workload. Then I needed to install the MFC components via the installer (Search in Individual Components)
Then when I tried to open the solution - I still got the typical errors. There are three errors, one for each generated vcxproj-file. Clicking on the error opens the corresponding file and there I made the following changes for each:
1) change the windows target platform from 8.1 to 10
2) change the Wwise version to the correct version (2023.1.0.8256 in my case)
3) change the <PlatformToolset> version from v140 to v143 (multiple occurences)
4) change all the occurences ending with '_vc140.props' to them ending with '_vc160.props'

Having done all that I could build the Release version (not sure about Debug) and it shows up and is audible and controllable in Wwise. 
Next I'm testing if it works in Godot.
I hope this is helpful!
Cheers

**Comment by O2xH** [comment]
I've created a ticket for Wwise generator upgrade #126

**Comment by eu-ch** [comment]
@mbbmbbmm we have upgraded Wwise support:
https://github.com/Wasted-Audio/hvcc/pull/130

Would be great if you could help us testing it -- please let me know if you have time for this :)

**Comment by eu-ch** [comment]
@mbbmbbmm Awesome, thanks for your time! Please check these temporary instructions: https://eugn.ch/hvcc_wwise_test.html -- let me know if you have any questions.
